tests/gui/test_widgets.py

```diff
@@ -143,6 +143,15 @@ def test_MainView_update_error_status(mocker):
     mv.error_status.setText.assert_called_once_with(expected_message)
 
 
+def test_MainView_re_create_conversation_view(mocker):
+    mv = MainView(None)
+    old_view_holder = mv.view_holder
+    old_view_layout = mv.view_layout
+    mv.re_create_conversation_view()
+    assert old_view_holder != mv.view_holder
+    assert old_view_layout != mv.view_layout
```

Comment:
Idea: Another nice test would be one for `_on_delete_source_complete` that covers the case where the user has clicked to source B while the deletion for source A was processing (i.e. the conversation view currently displays the conversation for the source that was _not_ just deleted).